openmls/src/group/public_group/validation.rs

```diff
@@ -308,6 +309,70 @@ impl PublicGroup {
         Ok(encryption_keys)
     }
 
+    /// Validate PreSharedKey proposals.
+    ///
+    /// This method implements the following checks:
+    ///
+    /// * ValSem401
+    /// * ValSem402
+    /// * ValSem403
+    pub(crate) fn validate_pre_shared_key_proposals(
+        &self,
+        proposal_queue: &ProposalQueue,
+    ) -> Result<(), ProposalValidationError> {
+        // ValSem403 (1/2)
+        // Note: Duplicate proposals are (likely) filtered. Let's do this check here anyway.
```

Comment:
If they are always filtered, we can skip this here.
I guess it depends on the definition of "duplicate". I'm fine with keeping this as is but maybe make a note in an issue (e.g. validation) to double check that we don't do unnecessary work.